modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
 // This module provides an array domain map class that is used to
 // represent rank change slices of another array via a domain.
 //
+// Throughout this file, 'down*' refers to the higher-dimensional
+// array/domain/distribution that's been rank-changed in this view,
+// while 'up*' refers to the lower-dimensional, user-facing view.
+//
+// TODOs for this file:
+//
+// * the dist, dom, and arr classes store a certain amount of
+//   duplicated information -- should these be focused in the higher
+//   classes (dist/dom) and referenced by lower ones (dom/dist)?
+//
+// * I don't think I've done anything to privatize the dist or dom
+//   classes... should I?
+//
 module ArrayViewRankChange {
 
   //
+  // This class represents a distribution that knows how to create
+  // rank-change domains and arrays similar to the one that caused
+  // it to be created.  It stores a pointer down to the distribution
+  // that it creates lower-dimensional views of, and two tuples
+  // indicating which dimensions were collapsed by the rank-change
+  // and what the indices of those dimensions are.
+  //
+  class ArrayViewRankChangeDist: BaseDist {
+    const downdist;
```

Comment:
The comment on the class does a good job of explaining this, but I think it'd be nice to have above-line or end-of-line comments on these variables to say what they are expected to be.